src/node_crypto.cc

```diff
@@ -1744,16 +1762,12 @@ static Local<Object> X509ToObject(Environment* env, X509* cert) {
       }
     }
 
-    unsigned char* pub = nullptr;
-    size_t publen = EC_KEY_key2buf(ec.get(), EC_KEY_get_conv_form(ec.get()),
-                                   &pub, nullptr);
-    if (publen > 0) {
-      Local<Object> buf = Buffer::New(env, pub, publen).ToLocalChecked();
-      // Ownership of pub pointer accepted by Buffer.
-      pub = nullptr;
+    const EC_POINT* pubkey = EC_KEY_get0_public_key(ec.get());
+    if (pubkey != nullptr) {
+      Local<Object> buf =
+          ECPointToBuffer(env, group, pubkey, EC_KEY_get_conv_form(ec.get()))
+              .ToLocalChecked();
```

Comment:
This will abort on failure. The idiomatic way of handling that is this:
```c++
Local<Object> buf;
if (ECPointToBuffer(...).ToLocal(&buf)) {
  // ...
} else {
  // Exception pending.
}
```
FWIW, I'm not sure it's a good idea to throw an exception from `ECPointToBuffer()` because that means `X509ToObject()` can leave the VM in an exception state, something its callers don't expect. Maybe you can pass the error message through an out param back to the caller?

Reply:
Done. (The distinction between two error cases probably doesn't matter, but I've preserved it, in case it does.)